Assets/Tests/InputSystem.Editor/XRIPackageTest.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEditor;
+using UnityEngine.TestTools;
+using UnityEditor.PackageManager;
+using UnityEditor.PackageManager.Requests;
+
+// Disable irrelevant warning about there not being underscores in method names.
+#pragma warning disable CA1707
+
+public class XRIPackageTests
+{
+    static AddRequest XRAddRequest;
+    static RemoveRequest XRRemoveRequest;
+
+    /// <summary>
+    /// TearDown removes the added XRI package again.
+    /// If you are adding a 2nd test that needs the XRI package adjust this.
+    /// </summary>
+    /// <returns></returns>
+    [UnityTearDown]
+    public IEnumerator TearDown()
+    {
+        XRRemoveRequest = Client.Remove("com.unity.xr.interaction.toolkit");
+        EditorApplication.update += RemoveProgress;
+        while (!XRRemoveRequest.IsCompleted)
+        {
+            yield return null;
+        }
+    }
+
+    [UnityTest]
+    [Category("Integration")]
+    public IEnumerator AdddingLatestXRIPackageThrowsNoErrors()
+    {
+        Application.logMessageReceived += HandleLog;
+
+        XRAddRequest = Client.Add("com.unity.xr.interaction.toolkit");
+        EditorApplication.update += AddProgress;
+
+        while (!XRAddRequest.IsCompleted)
+        {
+            yield return null;
+        }
+
+        AssetDatabase.Refresh();
+
+        yield return new WaitForDomainReload();
```

Comment:
Tests should assert something. So perhaps you should check that the log doesn't contain any errors

Reply:
Oh, I see assertion is hidden inside HaldleLog, in such case I would suggest writing all log messages into List and assert on them at the end of the test.